### PR TITLE
Propose dependency changes

### PR DESCRIPTION
**Pull request by florian-huber**
- Python from 3.8 - 3.11 changed to 3.9 to 3.12
- updated dependency versions

### CODE REVIEW

File: pyproject.toml
Comment (hechth) [inline pyproject.toml:1]: @florian-huber this seems to be invalid syntax to me?

File: pyproject.toml [hunk 1]
```diff
@@ -24,23 +24,23 @@ classifiers = [
...
-deprecated = "^1.2.14"
+matplotlib = ">=3.7"
+networkx = ">=3.0"
+numba = ">=0.57.1,<0.59"
+pickydict = ">=0.4.0"
+pyteomics = ">=4.6"
+requests = ">=2.31.0"
+sparsestack = ">=0.4.1"
+tqdm = ">=4.65.0"
+rdkit = ">2023.3.2"
```
Comment (hechth) [inline pyproject.toml]:
```suggestion
rdkit = ">=2023.9.6"
```

**Comment by hechth** [comment]
blocked by https://github.com/kuelumbus/rdkit-pypi/issues/99

File: pyproject.toml
Comment (florian-huber) [reply pyproject.toml:1]: Which part do you mean?

**Comment by julianpollmann** [comment]
@florian-huber this can be merged

File: matchms/Scores.py [hunk 2]
```diff
@@ -206,8 +206,13 @@ def scores_by_reference(self, reference: ReferencesType,
...
         if name is None and len(self.score_names) > 1 and sort is True:
             raise IndexError("For sorting, score must be specified")
-        assert reference in self.references, "Given input not found in references."
-        selected_idx = int(np.where(self.references == reference)[0])
+
+        idx_array = np.where(self.references == reference)[0]
+        if idx_array.size > 0:
+            selected_idx = int(idx_array[0])
+        else:
+            raise ValueError("Given input not found in references.")
```
Comment (florian-huber) [inline matchms/Scores.py:214]:
```suggestion
        raise ValueError("Given input not found in references.")
```

File: matchms/filtering/SpectrumProcessor.py [hunk 3]
```diff
@@ -318,7 +318,9 @@ def to_dataframe(self):
             if filter_name not in processing_report["filter"].values:
                 processing_report.loc[len(processing_report)] = {"filter": filter_name}
 
-        processing_report = processing_report.set_index("filter").fillna(0)
+        with pd.option_context("future.no_silent_downcasting", True):
```
Comment (florian-huber) [inline matchms/filtering/SpectrumProcessor.py]: This seems to only work for Pandas >= 2.2
I got errors for Pandas 2.0.* and 2.1.*

Comment (florian-huber) [reply matchms/filtering/SpectrumProcessor.py]: Maybe add a try-except or check for the present pandas version?

**Comment by florian-huber** [comment]
@julianpollmann looks like this works now, thanks!

**Pull request merged**